### PR TITLE
Add node tests

### PR DESCRIPTION
**Pull request by Fullchee**
### Why this PR exists
Regression tests for Node functionality.

I created separate tests for Hybrid Nodes and Course Nodes.

### CODE REVIEW

File: js/components/graph/__tests__/Node.test.js [hunk 1]
```diff
@@ -94,12 +138,130 @@ describe("Course Node", () => {
...
-    const courseTextNode = graph.getByText("AAA100");
-    expect(courseTextNode.parentNode.id.toUpperCase()).toBe(
-      courseTextNode.innerHTML
-    );
+    const aaa100 = graph.getByText("AAA100").parentNode;
+    expect(aaa100.classList.contains("node")).toBe(true);
+  });
+
+  describe("Unselected Course Node", async () => {
+    it("should be 'takeable' if pre-reqs met and 'inactive' otherwise", async () => {
```
Comment (david-yz-liu) [inline js/components/graph/__tests__/Node.test.js]: Split this up into two separate tests.

File: js/components/graph/__tests__/Node.test.js [hunk 2]
```diff
@@ -50,9 +51,52 @@ describe("Hybrid Node", () => {
         onKeyDown: jest.fn()
       }
     };
-    const wrapper = shallow(<Node {...nodeProps} />);
+    const wrapper = shallow(<Node {...hybridNodeProps} />);
     expect(wrapper).toMatchSnapshot();
   });
+
+  it("should have the 'hybrid CSS class", async () => {
```
Comment (david-yz-liu) [inline js/components/graph/__tests__/Node.test.js]: There's a missing single-quote (`'hybrid'`)

File: js/components/graph/__tests__/Node.test.js [hunk 3]
```diff
@@ -50,9 +51,52 @@ describe("Hybrid Node", () => {
...
+    // need to convert the classList into an object on order to use Jest's compare
+    expect({ ...hybrid101.classList }).toEqual(cssClassDeepCopy);
+    fireEvent.click(hybrid101);
+    expect({ ...hybrid101.classList }).toEqual(cssClassDeepCopy);
+    fireEvent.mouseOver(hybrid101);
+    expect({ ...hybrid101.classList }).toEqual(cssClassDeepCopy);
+    fireEvent.mouseOut(hybrid101);
+    expect({ ...hybrid101.classList }).toEqual(cssClassDeepCopy);
+  });
+  it("should be 'active' when its pre-req parent is met, and 'inactive' otherwise", async () => {
```
Comment (david-yz-liu) [inline js/components/graph/__tests__/Node.test.js]: Use "prereq" instead of "pre-req" (globally, but I'm just mentioning it once).

**Comment by Fullchee** [comment]
Thank you for the feedback! I've made the requested changes and I'll work on the helper for the next PR.

**Pull request merged**